src/main/java/org/assertj/vavr/api/AbstractSetAssert.java

```diff
@@ -0,0 +1,19 @@
+package org.assertj.vavr.api;
+
+import io.vavr.collection.Set;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.EnumerableAssert;
+
+import java.util.Comparator;
+
+abstract class AbstractSetAssert<SELF extends AbstractSetAssert<SELF, ACTUAL, ELEMENT, ELEMENT_ASSERT>,
+        ACTUAL extends Set<? extends ELEMENT>,
+        ELEMENT,
+        ELEMENT_ASSERT extends AbstractAssert<ELEMENT_ASSERT, ELEMENT>>
+        extends AbstractTraversableAssert<SELF, ACTUAL, ELEMENT, ELEMENT_ASSERT>
+        implements EnumerableAssert<SELF, ELEMENT> {
```

Comment:
I guess it cannot be implemented in a different way but it's very difficult to read.

Reply:
I agree!